chainstate/src/detail/transaction_verifier/error.rs

```diff
@@ -101,35 +86,60 @@ impl From<chainstate_storage::Error> for ConnectTransactionError {
     }
 }
 
-impl From<SpendError> for ConnectTransactionError {
+impl From<SpendError> for TxIndexError {
     fn from(err: SpendError) -> Self {
         match err {
-            SpendError::AlreadySpent(spender) => {
-                ConnectTransactionError::DoubleSpendAttempt(spender)
-            }
-            SpendError::AlreadyUnspent => ConnectTransactionError::InvariantBrokenAlreadyUnspent,
+            SpendError::AlreadySpent(spender) => TxIndexError::DoubleSpendAttempt(spender),
+            SpendError::AlreadyUnspent => TxIndexError::InvariantBrokenAlreadyUnspent,
             SpendError::OutOfRange {
                 tx_id,
                 source_output_index,
-            } => ConnectTransactionError::OutputIndexOutOfRange {
+            } => TxIndexError::OutputIndexOutOfRange {
                 tx_id,
                 source_output_index,
             },
         }
     }
 }
 
-impl From<TxMainChainIndexError> for ConnectTransactionError {
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+pub enum TxIndexError {
```

Comment:
Moved errors from `ConnectTransactionError` to a separate enum to break circular dependency between `TransactionVerifierError` and `ConnectTransactionError`